ftc/agents/lqr_tracking.py

```diff
@@ -0,0 +1,264 @@
+                      [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0],
+                      [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1],
+                      [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+                      [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+                      [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]])
+        B = np.array([[0, 0, 0, 0],
+                      [0, 0, 0, 0],
+                      [0, 0, 0, 0],
+                      [0, 0, 0, 0],
+                      [0, 0, 0, 0],
+                      [-1/self.m, 0, 0, 0],
+                      [0, 0, 0, 0],
+                      [0, 0, 0, 0],
+                      [0, 0, 0, 0],
+                      [0, self.Jinv[0, 0], 0, 0],
+                      [0, 0, self.Jinv[1, 1], 0],
+                      [0, 0, 0, self.Jinv[2, 2]]])
+        C = np.array([[1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+                      [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+                      [0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+                      [0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0],
+                      [0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0],
+                      [0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0]])
+        Q = np.diag([5, 5, 5, 0, 0, 0, 5, 5, 5, 0, 0, 0])
+        R = np.diag([1, 1, 1, 1])
+        self.K, *_ = LQR.clqr(A, B, Q, R)
+
+    def deriv(self, pos, vel, quat, omega):
+        M = np.vstack((self.M1, self.M2, self.M3))
+
+        m, g, J = self.m, self.g, self.J
+        e3 = np.vstack((0, 0, 1))
+
+        dpos = vel
+        dcm = quat2dcm(quat)
+        dvel = g*e3 - self.F*dcm.dot(e3)/m
+        dquat = 0.5 * np.vstack((
+            -omega.T.dot(quat[1:]),
+            omega*quat[0] - np.cross(omega, quat[1:], axis=0)
+        ))
+        domeg = self.Jinv.dot(M - np.cross(omega, J.dot(omega), axis=0))
+
+        return dpos, dvel, dquat, domeg
+
+    def set_dot(self, t):
+        states = self.observe_list()
+        dots = self.deriv(*states)
+        self.pos.dot, self.vel.dot, self.quat.dot, self.omega.dot = dots
+
+    def step(self):
+        t = self.clock.get()
+        states = self.observe_dict()
+        pos = self.pos.state
+        vel = self.vel.state
+        quat = self.quat.state
+        angle = np.array([np.flip(quat2angle(quat))]).reshape((3, 1))
+        omega = self.omega.state
+
+        x = np.vstack((pos, vel, angle, omega)) - self.ref
+        self.u = -self.K.dot(x) + np.vstack([self.m*self.g, 0, 0, 0])
+        self.F, self.M1, self.M2, self.M3 = self.u
+
+        rotors = self.mixer(self.u)
+
+        *_, done = self.update()
+        return t, states, rotors, done
+
+
+def run(pos, quat, ref):
+    system = Multicopter(pos=pos, quat=quat, ref=ref)
+    system.reset()
+    logger = fym.logging.Logger(path='data.h5')
+
+    while True:
+        system.render()
+        t, states, rotors, done = system.step()
+        logger.record(t=t, **states, rotors=rotors)
+
+        if done:
+            break
+
+    system.close()
```

Comment:
```suggestion
    system.close()
    logger.close()
```

`logger.close()`를 해주어야 합니다. `system.logger`로 등록하고, `system.logger_callback`을 작성하지 않는 이상